### PR TITLE
feat: adds opensearchservice

### LINKED ISSUE #1228: Opensearch Support
### What resource do you need?
OpenSearch (formerly ElasticSearch)
https://aws.amazon.com/opensearch-service/the-elk-stack/what-is-opensearch/

### What is your use case?
This is one of the last use cases needed for our workloads' infrastructure needs in order to have a complete crossplane adoption. One of it's current uses is as a caching layer in front of mongodb datastore.

### Would you be willing to contribute it using [code generator](https://github.com/crossplane/provider-aws/blob/master/CODE_GENERATION.md)?
I can try in my free time, although I've not worked in GO yet. 

Opening this request based on suggestion found in this issue:
https://github.com/crossplane/provider-aws/issues/238#issuecomment-1073776452

@haarchri @sqerison

### PR DESCRIPTION
**Pull request by jcooklin**
### Description of your changes

Adds open search service

Fixes #1228

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Create/Delete have been tested successfully in us-west-2 using examples/opensearchservice/domain.yaml

### CODE REVIEW

**Comment by haarchri** [comment]
Can you Check DCO Signing ? ;)

File: apis/opensearchservice/generator-config.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+          input_fields:
+            DomainName: Name
+        DescribeDomain:
+          input_fields:
+            DomainName: Name
+    reconcile:
+      # Doing this because it takes a LONG time for the Domain's
+      # endpoint/endpoints field to be populated, even after the Domain's
+      # Processing field is set to False after creation...
+      requeue_on_success_seconds: 60
```
Comment (chlunde) [inline apis/opensearchservice/generator-config.yaml:30]: is this (and the hooks) in use?

**Comment by chlunde** [comment]
cc @mvkrishna86 @sendmars - looks like all of you are interested in this, if you can, please coordinate. I think we have two PRs now.

File: apis/opensearchservice/generator-config.yaml (hunk 1 above)
Comment (jcooklin) [reply apis/opensearchservice/generator-config.yaml]: Nope.  I removed both of them.  

They were copied over from the ack controller but are not used in the case of Crossplane code gen.

**Review by haarchri — APPROVED** [review]
LGTM thanks

**Comment by haarchri** [comment]
@jcooklin can you Check for merge conflicts?

**Comment by jcooklin** [comment]
@haarchri ✅

**Pull request merged**